Swap assignment not working in closure

### Describe the bug

Cannot use `a, b = b, a` in a closure.

### Reproduction Steps

```v
module main

fn main() {
	f := foo()
	f()
}

fn foo() fn () {
	return fn () {
		mut left := 1
		mut right := 2

		left, right = right, left

		println('${left} ${right}')
	}
}
```

### Expected Behavior

No compiler error.

### Current Behavior

Produces a C compiler error:

```
==================
/tmp/v_501/bug.3024684967219090772.tmp.c:2081:17: error: use of undeclared identifier '_V_closure_ctx'
        int _var_114 = _V_closure_ctx->left;
                       ^
/tmp/v_501/bug.3024684967219090772.tmp.c:2082:17: error: use of undeclared identifier '_V_closure_ctx'
        int _var_120 = _V_closure_ctx->right;
                       ^
2 errors generated.
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.
```

### Possible Solution

You can work around this with a temporary variable:

```v
tmp := left
left = right
right = left
```

### Additional Information/Context

_No response_

### V version

V 0.4.2 4bc9a8f.48a1d6c

### Environment details (OS name and version, etc.)

```
V full version: V 0.4.2 4bc9a8f.48a1d6c
OS: macos, macOS, 12.6, 21G115
Processor: 10 cpus, 64bit, little endian, Apple M1 Max

getwd: /Users/elliot
vexe: /Users/elliot/Downloads/v-5/v
vexe mtime: 2023-10-21 03:59:08

vroot: OK, value: /Users/elliot/Downloads/v-5
VMODULES: OK, value: /Users/elliot/.vmodules
VTMP: OK, value: /tmp/v_501

Git version: git version 2.37.0 (Apple Git-136)
Git vroot status: weekly.2023.42-19-g48a1d6cc (94 commit(s) behind V master)
.git/config present: true

CC version: Apple clang version 14.0.0 (clang-1400.0.29.102)
thirdparty/tcc status: thirdparty-macos-amd64 46662e20
```
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/assign.v
+++ b/vlib/v/gen/c/assign.v
@@ -891,7 +891,13 @@ fn (mut g Gen) gen_cross_var_assign(node &ast.AssignStmt) {
 			ast.Ident {
 				left_typ := node.left_types[i]
 				left_sym := g.table.sym(left_typ)
-				anon_ctx := if g.anon_fn { '${closure_ctx}->' } else { '' }
+				mut anon_ctx := ''
+				if g.anon_fn {
+					obj := left.scope.find(left.name)
+					if obj is ast.Var && obj.is_inherited {
+						anon_ctx = '${closure_ctx}->'
+					}
+				}
 				if left_sym.kind == .function {
 					g.write_fn_ptr_decl(left_sym.info as ast.FnType, '_var_${left.pos.pos}')
 					g.writeln(' = ${anon_ctx}${c_name(left.name)};')

--- a/vlib/v/tests/closure_test.v
+++ b/vlib/v/tests/closure_test.v
@@ -223,3 +223,23 @@ fn test_closure_over_variable_that_is_returned_from_a_multi_value_function() {
 	a()
 	println(two)
 }
+
+fn test_cross_var_assign_without_inherited() {
+	f := fn () {
+		mut left := 1
+		mut right := 2
+		left, right = right, left
+		assert left == 2 && right == 1
+	}
+	f()
+}
+
+fn test_cross_var_assign_with_inherited() {
+	mut left := 1
+	mut right := 2
+	f := fn [mut left, mut right] () {
+		left, right = right, left
+		assert left == 2 && right == 1
+	}
+	f()
+}

